chia/wallet/puzzles/load_clvm.py

```diff
@@ -1,53 +1,106 @@
 import importlib
 import inspect
 import os
+import time
+
+from typing import Optional, TextIO
 import pathlib
 
 import pkg_resources
-from clvm_tools.clvmc import compile_clvm as compile_clvm_py
 from chia.types.blockchain_format.program import Program, SerializedProgram
+from clvm_tools_rs import compile_clvm as compile_clvm_rust
+
+
+# Cribbed mostly from chia/daemon/server.py
+def create_exclusive_lock(lockfile) -> Optional[TextIO]:
+    """
+    Open a lockfile exclusively.
+    """
+
+    try:
+        fd = os.open(lockfile, os.O_CREAT | os.O_EXCL | os.O_RDWR)
+        f = open(fd, "w")
 
-compile_clvm = compile_clvm_py
+        f.write("lock")
+    except IOError:
+        return None
 
-# Handle optional use of clvm_tools_rs if available and requested
-if "CLVM_TOOLS_RS" in os.environ:
+    return f
+
+
+compile_clvm_py = None
+
+
+def translate_path(p_):
+    p = str(p_)
+    if os.path.isdir(p):
+        return p
+    else:
+        module_object = importlib.import_module(p)
+        return os.path.dirname(inspect.getfile(module_object))
+
+
+# Handle optional use of python clvm_tools if available and requested
+if "CLVM_TOOLS" in os.environ:
     try:
+        from clvm_tools.clvmc import compile_clvm as compile_clvm_py_candidate
 
+        compile_clvm_py = compile_clvm_py_candidate
```

Comment:
If it is possible, it would be nice to import the version of the python clvm module. Strictly nice-to-have.

Reply:
Do i understand right that you're asking to also import python clvm alongside python clvm_tools?

Reply:
No, I just mean to be able to say "hey, I tried clvm_tools version x.y.z" in a diagnostic.
As even though the `clvm_tools` dependency on `clvm` is `>=`, having the version in a failure diagnostic will put them on the right track if they need to update.